src/main/java/games/strategy/engine/pbem/TripleAForumPoster.java

```diff
@@ -0,0 +1,202 @@
+package games.strategy.engine.pbem;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import games.strategy.debug.ClientLogger;
+import games.strategy.engine.framework.system.HttpProxy;
+import games.strategy.net.OpenFileUtility;
+import games.strategy.triplea.help.HelpSupport;
+import games.strategy.util.Util;
+
+public class TripleAForumPoster extends AbstractForumPoster {
+
+  private static final long serialVersionUID = -3380344469767981030L;
+
+  public static final String tripleAForumURL = "https://forums.triplea-game.org";
+
+  private NameValuePair username;
+  private NameValuePair password;
+
+
+  @Override
+  public boolean postTurnSummary(String summary, String title) {
+    username = new BasicNameValuePair("username", getUsername());
+    password = new BasicNameValuePair("password", getPassword());
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+      int userID = getUserId(client);
+      String token = getToken(client, userID);
+      try {
+        post(client, token, "### " + title + "\n" + summary);
+        display("Sucessfully posted!");
+        return true;
+      } finally {
+        deleteToken(client, userID, token);
+      }
+    } catch (JSONException e) {
+      ClientLogger.logError("Invalid JSON", e);
```

Comment:
Pretty redundant error handling can the exception blocks be merged or consolidated to a common method?